src/vm/amd64/asmhelpers.S

```diff
@@ -0,0 +1,196 @@
+// SIZEOF_STACK_FRAME
+
+.equ OFFSETOF_PLATFORM_SPECIFIC_DATA, SIZEOF_OUTGOING_ARGUMENT_HOMES
+
+// we'll just spill into the PROFILE_PLATFORM_SPECIFIC_DATA structure
+.equ OFFSETOF_FP_ARG_SPILL, SIZEOF_OUTGOING_ARGUMENT_HOMES + SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA
+
+.equ SIZEOF_STACK_FRAME, SIZEOF_OUTGOING_ARGUMENT_HOMES + SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA + SIZEOF_MAX_FP_ARG_SPILL
+
+.equ PROFILE_ENTER, 0x1
+.equ PROFILE_LEAVE, 0x2
+.equ PROFILE_TAILCALL, 0x4
+
+// ***********************************************************
+//   NOTE:
+//
+//   Register preservation scheme:
+//
+//       Preserved:
+//           - all non-volatile registers
+//           - rax
+//           - xmm0
+//
+//       Not Preserved:
+//           - integer argument registers (rcx, rdx, r8, r9)
+//           - floating point argument registers (xmm1-3)
+//           - volatile integer registers (r10, r11)
+//           - volatile floating point registers (xmm4-5)
+//           - upper halves of ymm registers on AVX (which are volatile)
+//
+// ***********************************************************
+
+// EXTERN_C void ProfileEnterNaked(FunctionIDOrClientID functionIDOrClientID, size_t profiledRsp);
+// <NOTE>
+//
+// </NOTE>
+NESTED_ENTRY ProfileEnterNaked, _TEXT, NoHandler
+  //       Upon entry :
+  //           rdi = clientInfo
+  //           rsi = profiledRsp
+
+  PUSH_CALLEE_SAVED_REGISTERS
+  push_argument_register  r9
+  push_argument_register  r8
+  push_argument_register  rcx
+  push_argument_register  rdx
+  push_nonvol_reg         r10
+  push_nonvol_reg         rax
+
+  lea                     rax, [rsp + 0x10]    // caller rsp
+  mov                     r10, [rax - 0x8]     // return address
+
+  alloc_stack             SIZEOF_STACK_FRAME
+
+  // correctness of return value in structure doesn't matter for enter probe
+
+
+  // setup ProfilePlatformSpecificData structure
+  xor                     r11, r11 // nullify r11
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA +  0x0], r11    // r11 is null      -- struct functionId field
+  save_reg_postrsp        rbp, OFFSETOF_PLATFORM_SPECIFIC_DATA +    0x8          //                 -- struct rbp field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x10], rax    // caller rsp      -- struct probeRsp field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x18], r10    // return address  -- struct ip field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x20], rsi    //                 -- struct profiledRsp field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x28], r11    // return value    -- struct rax field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x30], r11    // r11 is null      -- struct hiddenArg field
+  movsd                   real8 ptr [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x38], xmm0    //      -- struct flt0 field
```

Comment:
vm\proftoeeinterfaceimpl.cpp - ProfilingGetFunctionEnter3Info function has more detail if you want to see it but yes, the basic idea is that a developer can call the profiler APIs to iterate through the arguments. The arguments are specified by their address in memory so stack homed arguments automatically have an address, but arguments that are register homed only need to be spilled somewhere. In the case of floating point arguments, they were spilled not to the stack but into this PLATFORM_SPECIFIC_DATA structure.